### PR TITLE
feat(client): adapter/cloudflare-d1 - `PrismaD1` (first draft)

### PR DESCRIPTION
**Pull request by Druue**
contributes https://github.com/prisma/team-orm/issues/830
closes https://github.com/prisma/team-orm/issues/834

(WIP)

### CODE REVIEW

**Review by Jolg42 — APPROVED** [review]
As discussed with @Druue we want to merge this and improve it in next PRs.

File: packages/adapter-d1/src/d1.ts [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+class D1Transaction extends D1Queryable<StdClient> implements Transaction {
+  constructor(client: StdClient, readonly options: TransactionOptions) {
+    super(client)
+  }
+
+  async commit(): Promise<Result<void>> {
+    console.debug(`[js::commit]`)
+
+    // TODO remove (added to have linting pass)
+    await new Promise((resolve) => resolve)
```
Comment (Druue) [inline packages/adapter-d1/src/d1.ts:151]: Note: this causes wrangler errors
image

File: packages/adapter-d1/src/d1.ts [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+
+    return ok(undefined)
+  }
+
+  async rollback(): Promise<Result<void>> {
+    console.debug(`[js::rollback]`)
+
+    // TODO remove (added to have linting pass)
+    await new Promise((resolve) => resolve)
```
Comment (Druue) [inline packages/adapter-d1/src/d1.ts:160]: and here

File: packages/adapter-d1/src/d1.ts [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+    const options: TransactionOptions = {
+      // TODO: D1 does not support transactions.
+      usePhantomQuery: true,
+    }
+
+    const tag = '[js::startTransaction]'
+    console.debug(`${tag} options: %O`, options)
+
+    // TODO remove (added to have linting pass)
+    await new Promise((resolve) => resolve)
```
Comment (Druue) [inline packages/adapter-d1/src/d1.ts:181]: and here

Comment (Jolg42) [reply packages/adapter-d1/src/d1.ts:181]: My fault 😄 
I think we can do better than that now 😉

**Pull request merged**